db/push.go

```diff
@@ -0,0 +1,96 @@
+package db
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	nomsjson "roci.dev/diff-server/util/noms/json"
+)
+
+// These request/response types might want to live in a different location, eg 
+// in diff-server/serve/types.
+type Mutation struct {
+	ID   uint64          `json:"id"`
+	Name string          `json:"name"`
+	Args json.RawMessage `json:"args"`
+}
+
+type BatchPushRequest struct {
+	ClientID  string     `json:"clientId"`
+	Mutations []Mutation `json:"mutations"`
+}
+
+type MutationInfo struct {
+	ID    string `json:"id"`
+	Error string `json:"error"`
```

Comment:
I think maybe we should rename this field `Message`. I imagine developers optionally returning various informational messages in it - not necessarily errors. Ane the client would just print it to the console.

Also, instead of returning an array of these, what if we had the response look like:

```
{
  messages: {
    "42": "server went bonk",
    "88": "conflict - time slot not available",
  }
}
```

Reply:
This way we don't have the server returning a bunch of empty objects in the common case it has nothing to say.